src/Misc/PerformanceViewer/performanceViewerCollector.ts

```diff
@@ -0,0 +1,225 @@
+     */
+    public readonly datasets: IPerfDatasets;
+    /**
+     * An observable you can attach to get deltas in the dataset. Subscribing to this will increase memory consumption slightly, and may hurt performance due to increased garbage collection needed.
+     */
+    public readonly datasetObservable: Observable<IPerfDataSliceSnapshot>;
+    /**
+     * An observable you can attach to get the most updated map of metadatas.
+     */
+    public readonly metadataObservable: Observable<Map<string, IPerfMetadata>>;
+
+    /**
+     * The offset for when actual data values start appearing inside a slice.
+     */
+    public static get SliceDataOffset() {
+        return 2;
+    }
+
+    /**
+     * The offset for the value of the number of points inside a slice.
+     */
+    public static get NumberOfPointsOffset() {
+        return 1;
+    }
+
+    /**
+     * Handles the creation of a performance viewer collector.
+     * @param _scene the scene to collect on.
+     * @param _enabledStrategyCallbacks the list of data to collect with callbacks for initialization purposes.
+     */
+    constructor(private _scene: Scene, _enabledStrategyCallbacks?: PerfStrategyInitialization[]) {
+        this.datasets = {
+            ids: [],
+            data: new DynamicFloat32Array(initialArraySize),
+            startingIndices: new DynamicFloat32Array(initialArraySize)
+        };
+        this._strategies = new Map<string, IPerfViewerCollectionStrategy>();
+        this._datasetMeta = new Map<string, IPerfMetadata>();
+        this.datasetObservable = new Observable();
+        this.metadataObservable = new Observable((observer) => observer.callback(this._datasetMeta, new EventState(0)));
+        if (_enabledStrategyCallbacks) {
+            this.addCollectionStrategies(_enabledStrategyCallbacks);
+        }
+    }
+
+    /**
+     * This method adds additional collection strategies for data collection purposes.
+     * @param strategyCallbacks the list of data to collect with callbacks.
+     */
+    public addCollectionStrategies(strategyCallbacks: PerfStrategyInitialization[]) {
+        for (const strategyCallback of strategyCallbacks) {
+            const strategy = strategyCallback(this._scene);
+
+            if (this._strategies.has(strategy.id)) {
+                continue;
+            }
+
+            this.datasets.ids.push(strategy.id);
+
+            this._datasetMeta.set(strategy.id, {
+                color: this._getHexFromId(strategy.id),
+            });
+
+            this._strategies.set(strategy.id, strategy);
+        }
+
+        this.metadataObservable.notifyObservers(this._datasetMeta);
+    }
+
+    /**
+     * Gets a 6 character hexcode from a passed in string.
+     * @param id the string to get a hex code for.
+     * @returns a hexcode hashed from the id.
+     */
+    private _getHexFromId(id: string) {

```

Comment:
is that just a way to hash the id, or is there a reason behind the 6 characters?

Reply:
Oh it gives you a 6 character hexcode representing the colour. Perhaps I should rename this to _getHexColorFromId?

Reply:
that would make it clearer. 
You could also assign a random color ;-) but it is nice that it is generated. this way every time a specific ID will be used it will have the same color.

Reply:
Changed the name!